spatialdata/_core/models.py

```diff
@@ -306,32 +305,75 @@ def validate(cls, data: GeoDataFrame) -> None:
             raise KeyError(f"GeoDataFrame must have a column named `{cls.GEOMETRY_KEY}`.")
         if not isinstance(data[cls.GEOMETRY_KEY], GeoSeries):
             raise ValueError(f"Column `{cls.GEOMETRY_KEY}` must be a GeoSeries.")
-        if not isinstance(data[cls.GEOMETRY_KEY].values[0], Polygon) and not isinstance(
-            data[cls.GEOMETRY_KEY].values[0], MultiPolygon
-        ):
-            # TODO: should check for all values?
-            raise ValueError(f"Column `{cls.GEOMETRY_KEY}` must contain Polygon or MultiPolygon objects.")
+        geom_ = data[cls.GEOMETRY_KEY].values[0]
+        if not isinstance(geom_, (Polygon, MultiPolygon, Point)):
+            raise ValueError(
+                f"Column `{cls.GEOMETRY_KEY}` can only contain `Point`, `Polygon` or `MultiPolygon` shapes, but it contains {type(geom_)}."
+            )
+        if isinstance(geom_, Point):
+            if cls.RADIUS_KEY not in data.columns:
+                raise ValueError(f"Column `{cls.RADIUS_KEY}` not found.")
         if cls.TRANSFORM_KEY not in data.attrs:
             raise ValueError(f":class:`geopandas.GeoDataFrame` does not contain `{TRANSFORM_KEY}`.")
 
     @singledispatchmethod
     @classmethod
     def parse(cls, data: Any, **kwargs: Any) -> GeoDataFrame:
+        """
+        Validate (or parse) shapes data.
+
+        Parameters
+        ----------
+        data
+            Data to parse:
+
+                - If :np:class:`numpy.ndarray`, it assumes the shapes are parsed as ragged array,
+                therefore additional arguments `offsets` and `geometry` must be provided
+                in case of (Multi)`Polygons`.
+                - if `Path` or `str`, it's read as a GeoJSON file.
+                - If :class:`geopandas.GeoDataFrame`, it's validated.
+
+            A `radius` array can also be passed to store the radius of the `Circles`.
+
+        geometry
+            Geometry type of the shapes. The following geometries are supported:
+
+                - 0: `Circles`
+                - 3: `Polygon`
+                - 6: `MultiPolygon`
```

Comment:
what are the 0, 3, and 6 here?

Reply:
thse are defined by `GeometryType` enum by shapely and are used to set the geometry type in the parser.